src/mixings/dirichlet_mixing.hpp

```diff
@@ -38,19 +39,31 @@ class DirichletMixing : public BaseMixing {
   //! \param card Cardinality of the cluster
   //! \param n    Total number of data points
   //! \return     Probability value
-  double mass_existing_cluster(const unsigned int card,
-                               const unsigned int n) const override {
-    return card / (n + state.totalmass);
+  double mass_existing_cluster(std::shared_ptr<BaseHierarchy> hier,
+                               const unsigned int n, bool log,
+                               bool propto) const override {
+    double out;
+    if (log)
+      out = hier->get_log_card() - propto ? 0 : std::log(n +state.totalmass);
+    else 
+      out = hier->get_card() / propto ? 1 : (n + state.totalmass);
+    return out;
```

Comment:
This block and the next one could use some brackets to make it clear what's going on 😅

Reply:
brackets around the if/else statements?

Reply:
Yes, round brackets around the compact a ? b : c constructs.

While we are at it, I'd also like curly brackets around the standard if-else, even if it's just one line each, just for consistency with the rest of the code